packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2104,3 +2139,165 @@ class InputDecoration {
+  /// [EdgeInsets.zero].
+  final EdgeInsets contentPadding;
+
+  /// Whether the decoration is the same size as the input field.
+  ///
+  /// A collapsed decoration cannot have [InputDecoration.labelText],
+  /// [InputDecoration.errorText], or an [InputDecoration.icon].
+  final bool isCollapsed;
+
+  /// The style to use for the [InputDecoration.prefixText].
+  ///
+  /// If null, defaults to the [hintStyle].
+  final TextStyle prefixStyle;
+
+  /// The style to use for the [InputDecoration.suffixText].
+  ///
+  /// If null, defaults to the [hintStyle].
+  final TextStyle suffixStyle;
+
+  /// The style to use for the [InputDecoration.counterText].
+  ///
+  /// If null, defaults to the [helperStyle].
+  final TextStyle counterStyle;
+
+  /// If true the decoration's container is filled with [fillColor].
+  ///
+  /// Typically this field set to true if [border] is
+  /// [const UnderlineInputBorder()].
+  ///
+  /// The decoration's container is the area which is filled if [isFilled] is
+  /// true and bordered per the [border]. It's the area adjacent to
+  /// [InputDecoration.icon] and above the widgets that contain
+  /// [InputDecoration.helperText], [InputDecoration.errorText], and
+  /// [InputDecoration.counterText].
+  ///
+  /// This property is false by default.
+  final bool filled;
+
+  /// The color to fill the decoration's container with, if [filled] is true.
+  ///
+  /// By default the fillColor is based on the current [Theme].
+  ///
+  /// The decoration's container is the area which is filled if [isFilled] is
+  /// true and bordered per the [border]. It's the area adjacent to
+  /// [InputDecoration.icon] and above the widgets that contain
+  /// [InputDecoration.helperText], [InputDecoration.errorText], and
+  /// [InputDecoration.counterText].
+  final Color fillColor;
+
+  /// The border to draw around the decoration's container.
+  ///
+  /// The decoration's container is the area which is filled if [isFilled] is
+  /// true and bordered per the [border]. It's the area adjacent to
+  /// [InputDecoration.icon] and above the widgets that contain
+  /// [InputDecoration.helperText], [InputDecoration.errorText], and
+  /// [InputDecoration.counterText].
+  ///
+  /// The default value of this property is `const UnderlineInputBorder()`.
+  ///
+  /// See also:
+  ///  * [UnderlineInputBorder], which draws a horizontal line at the
+  ///    bottom of the input decorator's container.
+  ///  * [OutlineInputBorder], an [InputDecorator] border which draws a
+  ///    rounded rectangle around the input decorator's container.
+  final InputBorder border;
+
+  /// Used by widgets like [TextField] and [InputDecorator] to create a new
+  /// [InputDecoration] with default values taken from this theme.
+  ///
+  /// Only null valued properties from [baseDecoration] are replaced
+  /// by the corresponding values from this theme.
+  InputDecoration inputDecoration({ InputDecoration baseDecoration }) {
```

Comment:
this might be clearer if it was an API on InputDecoration itself, e.g. InputDecoration.applyDefaults(InputDecorationTheme)

Reply:
That's a good idea.